partiql-lang/src/main/kotlin/org/partiql/lang/eval/builtins/DynamicLookupExprFunction.kt

```diff
@@ -105,3 +102,13 @@ class DynamicLookupExprFunction : ExprFunction {
             }
         }.firstOrNull { it != null }
 }
+
+class DynamicLookupExprFunction2 : ExprFunction {
+
+    override val signature = FunctionSignature(
+        name = DYNAMIC_LOOKUP_FUNCTION_NAME,
+        // Required parameters are: variable name, case sensitivity and lookup strategy
+        requiredParameters = listOf(StaticType.SYMBOL, StaticType.SYMBOL, StaticType.SYMBOL),
+        returnType = StaticType.ANY
+    )
```

Comment:
Seemingly, we need to still look in the session's globals:
```kotlin
val found = when {
            globalsFirst -> {
                session.globals[bindingName] ?: searchLocals(variadic, bindingName)
            }
            else -> {
                searchLocals(variadic, bindingName) ?: session.globals[bindingName]
            }
        }

        if (found == null) {
            // We don't know the metas inside ExprFunction implementations.  The ThunkFactory error handlers
            // should add line & col info to the exception & rethrow anyway.
            throwUndefinedVariableException(bindingName, metas = null)
        } else {
            return found
        }
```

And we don't need to search through the locals (`searchLocals`). But, if we can't find in the globals, we still need to throw that same error.